### PR TITLE
refactor!: Replace unmaintained derivative create with educe

### PR DESCRIPTION
**Pull request by sbernauer**
# Description

Fixes https://github.com/stackabletech/operator-rs/issues/905
Part of https://github.com/stackabletech/issues/issues/677

Also updated `deny.toml` to match the one in operator-templating.

We still have the `instant` and `rsa` crates, but that will be separate PRs.

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes

```[tasklist]
# Author
- [x] Changes are OpenShift compatible
- [x] CRD changes approved
- [ ] Integration tests passed (for non trivial changes)
```

```[tasklist]
# Reviewer
- [x] Changelog updated
- [x] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
```

### CODE REVIEW

**Review by siegfriedweber — CHANGES_REQUESTED** [review]

File: crates/stackable-operator/src/crd.rs [hunk 1]
```diff
@@ -1,19 +1,12 @@
...
 ///
 /// `namespace`'s defaulting only applies when retrieved via [`ClusterRef::namespace_relative_from`]
-#[derive(Deserialize, Serialize, JsonSchema, Derivative)]
-#[derivative(
-    Default(bound = ""),
-    Clone(bound = ""),
-    Debug(bound = ""),
-    PartialEq(bound = "")
-)]
+#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize, JsonSchema)]
```
[the remark this answers is not in the document]
Comment (sbernauer) [reply crates/stackable-operator/src/crd.rs]: Thanks, I got `cargo expand` working again on my machine and looked at the generated code.
Applied your suggestion in 9e98541

File: deny.toml [hunk 2]
```diff
@@ -7,49 +8,44 @@ targets = [
...
     "ISC",
     "LicenseRef-ring",
     "LicenseRef-webpki",
     "MIT",
+    "MPL-2.0",
+    "Unicode-3.0",
     "Unicode-DFS-2016",
     "Zlib",
     "Unlicense",
-    "OpenSSL",
```
[the remark this answers is not in the document]
Comment (sbernauer) [reply deny.toml:40]: I would prefer merging it as-is (so current state from op-templating) and raise an issue with our license-expert @lfrancke, so that we can allow this in op-templating and here at the same time

[the remark this answers is not in the document]
Comment (sbernauer) [reply deny.toml:40]: I think operator-templating should be the source of truth for the deny.toml and where license discussions should happen

[the remark this answers is not in the document]
Comment (sbernauer) [reply deny.toml:40]: Created https://github.com/stackabletech/operator-templating/pull/464 for the inclusion. If we merge it in time, we can directly included that in this PR

[the remark this answers is not in the document]
Comment (soenkeliebau) [reply deny.toml:40]: I'll take a quick look

[the remark this answers is not in the document]
Comment (soenkeliebau) [reply deny.toml:40]: Probably a dumb question, but why are there so many changes to the deny.toml in this PR?

Is this just to get the file in line with https://github.com/stackabletech/operator-templating/pull/430 in operator-templating?

[the remark this answers is not in the document]
Comment (soenkeliebau) [reply deny.toml:40]: In principle the OpenSSL license seems to be non-critical, but it does expect us to credit the openssl project in advertising ..

At the very least we need to add it to the licenses folder when building the docker images for our operators (do we even package licenses there yet?)..

https://www.tldrlegal.com/license/openssl-license-openssl
https://opensource.stackexchange.com/questions/11558/what-needs-to-be-done-to-comply-with-openssl-double-license

[the remark this answers is not in the document]
Comment (soenkeliebau) [reply deny.toml:40]: So, looks like the license was greenlit based on this comment: https://github.com/stackabletech/operator-rs/pull/753#issuecomment-2015244865 back in March.

The check now fails because @sbernauer changed the deny.toml to look like what we have in templating and the OpenSSL license wasn't added there back then, so now it was removed here.

I'd suggest adding it and also adding it in templating, just as @sbernauer proposed.

[the remark this answers is not in the document]
Comment (sbernauer) [reply deny.toml:40]: @Techassi any opinions on the use of aws-lc or ring?

[the remark this answers is not in the document]
Comment (sbernauer) [reply deny.toml:40]: Spoke with @Techassi, he's fine with either ring or aws-lc-rs

[the remark this answers is not in the document]
Comment (NickLarsenNZ) [reply deny.toml:40]: Ok,somewhat relevant to https://github.com/stackabletech/operator-rs/pull/891

[the remark this answers is not in the document]
Comment (NickLarsenNZ) [reply deny.toml:40]: Regarding ring, I'm keen for it, but I think since it is crypto, we maybe need some extra eyes across it to judge its fitness.

[the remark this answers is not in the document]
Comment (siegfriedweber) [reply deny.toml:40]: The aws_lc_rs provider uses ring. This discussion is only about the configuration of ring (which ciphers to include).

File: Cargo.toml [hunk 3]
```diff
@@ -19,9 +19,9 @@ const-oid = "0.9.6"
 convert_case = "0.6.0"
 darling = "0.20.10"
 delegate = "0.13.0"
-derivative = "2.2.0"
 dockerfile-parser = "0.8.0"
 ecdsa = { version = "0.16.9", features = ["digest", "pem"] }
+educe =  { version = "0.6.0", default-features = false, features = ["Clone", "Debug", "Default", "PartialEq"] }
```
[the remark this answers is not in the document]
Comment (siegfriedweber) [reply Cargo.toml:24]: educe is already used by kube-rs (https://github.com/kube-rs/kube/blob/0.97.0/Cargo.toml#L45), so we are not adding a new dependency. See also their pull request and its description: https://github.com/kube-rs/kube/pull/1585

File: deny.toml (hunk 2 above)
[the remark this answers is not in the document]
Comment (sbernauer) [reply deny.toml:40]: Actually the motivation of using ring over aws-lc-rs might be the OpenSSL license (which aws-lc-rs has).
However, looking at https://github.com/briansmith/ring/blob/main/LICENSE, ring also has parts below OpenSSL license, so that argument seems to be misleading. Looks like we need the OpenSSL license in any case, and we are just "lucky", that the ring crate does [not have machine readable license](https://github.com/briansmith/ring/blob/befdc87ac7cbca615ab5d68724f4355434d3a620/Cargo.toml#L8) compared to [aws-lc-sys](https://github.com/aws/aws-lc-rs/blob/cdb1651d7398148688fa4c0106135b4a82ec1e49/aws-lc-sys/Cargo.toml#L9).

So to sum up: I think ring isn't any better than aws-lc-rs from a license perspective. I would prefer to focus this PR on derivative and get it merged and do the ring vs aws-lc-rs discussion in https://github.com/stackabletech/decisions/issues/35

File: Cargo.toml (hunk 3 above)
[the remark this answers is not in the document]
Comment (sbernauer) [reply Cargo.toml:24]: What @siegfriedweber said + https://github.com/stackabletech/operator-rs/issues/905 suggested 3 crates.
`derive_more` has the most downloads, but I did not get it to work. `educe` was basically a drop-in replacement.

**Comment by sbernauer** [comment]
@siegfriedweber given https://github.com/stackabletech/operator-rs/pull/907/#discussion_r1856076753, are we ready to merge this and follow-up on the OpenSSL license in https://github.com/stackabletech/decisions/issues/35?

**Review by siegfriedweber — APPROVED** [review]
